File: desktop-app/test_epg.py
import os
import re

def test_epg():
    log = []
    log.append("--- TEST START ---")
    xml_path = "epg_test.xml"
    if not os.path.exists(xml_path):
        log.append(f"Archivo no encontrado: {xml_path}")
        with open('epg_debug_py.txt', 'w', encoding='utf-8') as f:
            f.write('\n'.join(log))
        return

    log.append(f"Archivo encontrado: {xml_path}")
    try:
        with open(xml_path, 'r', encoding='utf-8', errors='ignore') as f:
            text = f.read()
            log.append(f"Tamaño: {len(text)} caracteres")
    except Exception as e:
        log.append(f"Error leyendo archivo: {e}")
        with open('epg_debug_py.txt', 'w', encoding='utf-8') as f:
            f.write('\n'.join(log))
        return

    # Parsear Canales
    id_to_names = {}
    channel_pieces = text.split('</channel>')
    for piece in channel_pieces:
        match = re.search(r'<channel([^>]+)>', piece)
        if not match:
            continue
        attr_text = match.group(1)
        content = piece[match.end():]
        id_match = re.search(r'id="([^"]*)"', attr_text)
        cid = id_match.group(1) if id_match else ""
        names = re.findall(r'<display-name[^>]*>(?:<!\[CDATA\[)?([\s\S]*?)(?:\]\]>)?<\/display-name>', content, re.IGNORECASE)
        id_to_names[cid] = [n.strip() for n in names]

    log.append(f"Canales mapeados: {len(id_to_names)}")

    # Parsear Programas
    pieces = text.split('</programme>')
    count = 0
    channel_ids_with_progs = set()

    for piece in pieces:
        match = re.search(r'<programme([^>]+)>', piece)
        if not match:
            continue
        attr_text = match.group(1)
        channel_match = re.search(r'channel="([^"]+)"', attr_text)
        if channel_match:
            channel_ids_with_progs.add(channel_match.group(1))
            count += 1

    log.append(f"Programas totales: {count}")
    log.append(f"Canales con programas: {len(channel_ids_with_progs)}")

    log.append("\n--- Búsqueda de Adult Swim ---")
    contains_adult = [cid for cid, names in id_to_names.items() if 'adult' in cid.lower() or any('adult' in n.lower() for n in names)]
    
    log.append(f"Canales con 'Adult': {' | '.join([f'{cid} ({id_to_names[cid]})' for cid in contains_adult])}")
    
    for cid in contains_adult:
        has_progs = cid in channel_ids_with_progs
        log.append(f"¿Tiene programas {cid}?: {has_progs}")

    log.append("--- TEST END ---")

    with open('epg_debug_py.txt', 'w', encoding='utf-8') as f:
        f.write('\n'.join(log))

test_epg()

File: desktop-app/test_epg.js
const fs = require('fs');

function test() {
    let log = "";
    try {
        log += "Iniciando prueba...\n";
        const text = fs.readFileSync('epg_test.xml', 'utf-8');
        log += `Tamaño del archivo: ${text.length} caracteres\n`;
        
        const idToNames = {};
        const channelPieces = text.split('</channel>');
        log += `Piezas de canal: ${channelPieces.length}\n`;
        
        for (const piece of channelPieces) {
            const match = piece.match(/<channel([^>]+)>/);
            if (!match) continue;
            const attrText = match[1];
            const content = piece.substring(match.index + match[0].length);
            const id = attrText.match(/id="([^"]*)"/)?.[1] || "";
            const names = [...content.matchAll(/<display-name[^>]*>(?:<!\[CDATA\[)?([\s\S]*?)(?:\]\]>)?<\/display-name>/gi)];
            idToNames[id] = names.map(n => n[1].trim());
        }

        log += `Canales mapeados: ${Object.keys(idToNames).length}\n`;

        const pieces = text.split('</programme>');
        log += `Piezas de programa: ${pieces.length}\n`;
        
        let count = 0;
        const channelIdsWithProgs = new Set();

        for (const piece of pieces) {
            const match = piece.match(/<programme([^>]+)>/);
            if (!match) continue;
            const attrText = match[1];
            const channelId = attrText.match(/channel="([^"]+)"/)?.[1];
            if (channelId) {
                channelIdsWithProgs.add(channelId);
                count++;
            }
        }

        log += `Programas totales: ${count}\n`;
        log += `Canales con programas: ${channelIdsWithProgs.size}\n`;
        
        const examples = [...channelIdsWithProgs].slice(0, 10);
        log += `Ejemplos: ${examples.join(', ')}\n`;
        
        log += "\n--- Búsqueda de Adult Swim ---\n";
        const containsAdult = Object.keys(idToNames).filter(id => id.toLowerCase().includes('adult') || idToNames[id].some(n => n.toLowerCase().includes('adult')));
        
        log += `Canales con 'Adult': ${containsAdult.map(id => `${id} (${idToNames[id].join(', ')})`).join(' | ')}\n`;
        
        for (const id of containsAdult) {
            log += `¿Tiene programas ${id}?: ${channelIdsWithProgs.has(id)}\n`;
        }

    } catch (e) {
        log += `Error: ${e.message}\n${e.stack}\n`;
    } finally {
        fs.writeFileSync('epg_debug.txt', log, 'utf-8');
    }
}

test();
